tool/tsh/kubectl.go

```diff
@@ -275,3 +313,178 @@ func getKubeClusterName(args []string, teleportClusterName string) (string, erro
 	}
 	return kubeName, nil
 }
+
+type kubeLocalProxyOpts struct {
+	// kubectlArgs is a list of command arguments passed in for `tsh kubectl`.
+	// used to decide if local proxy is required.
+	kubectlArgs []string
+	// makeAndStartKubeLocalProxyFunc is a callback function to create and
+	// start a kube local proxy, when it is decided that a local proxy is
+	// required. Default to makeAndStartKubeLocalProxy. Can be set another
+	// function for testing.
+	makeAndStartKubeLocalProxyFunc func(*CLIConf, *clientcmdapi.Config, kubeconfig.LocalProxyClusters) (func(), string, error)
+	// onNewKubeconfigFuncs is list of callback functions that will receive the
+	// new kubeconfig location of the local proxy when the local proxy is
+	// successfully created.
+	onNewKubeconfigFuncs []func(string) error
+}
+
+type applyKubeLocalProxyOpts func(o *kubeLocalProxyOpts)
+
+func withKubectlArgs(args []string) applyKubeLocalProxyOpts {
+	return func(o *kubeLocalProxyOpts) {
+		o.kubectlArgs = args
+		o.onNewKubeconfigFuncs = append(o.onNewKubeconfigFuncs, func(newPath string) error {
+			overwriteKubeconfigFlagInArgs(args, newPath)
+			return nil
+		})
+	}
+}
+func withKubeconfigEnvUpdate() applyKubeLocalProxyOpts {
+	return func(o *kubeLocalProxyOpts) {
+		o.onNewKubeconfigFuncs = append(o.onNewKubeconfigFuncs, func(newPath string) error {
+			return trace.Wrap(os.Setenv(teleport.EnvKubeConfig, newPath))
```

Comment:
I am afraid that this might be problematic when running multiple commands in parallel since each command can override each others. Can we load the kubeconfig without the env var?

For `tsh kubectl` this works because you can set the value in the `exec.CMD` and it won't override the system value but for all other commands, it will be a  problem.